### ISSUE TITLE
negative weighting

### ISSUE THREAD

**Issue by nicolai256** [bloc97/CrossAttentionControl#10]
does negative weighting work properly?
most repos that have weighting don't have the option for negative weights,
does it work in this one?
thanks

**Comment by bloc97** [comment]
Yes, to my knowledge, only attention weighting, CFG weighting and CLIP guidance weighting works correctly with StableDiffusion. Changing the input embeddings by averaging or doing arithmetic on them is difficult as usually two prompts have different token lengths and usually breaks the generated image.

In this method only attention weighting is implemented, but I plan to add CFG weighting too (however it is slower than this method). CLIP weighting is the slowest so it might not be worth the effort for now.
